cluster/addons/rbac/cluster-autoscaler/cluster-autoscaler-rbac.yaml

```diff
@@ -0,0 +1,64 @@
+kind: ClusterRole
+apiVersion: rbac.authorization.k8s.io/v1
+metadata:
+  name: cluster-autoscaler
+  labels:
+    addonmanager.kubernetes.io/mode: Reconcile
+rules:
+  # leader election
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    verbs: ["create"]
+  - apiGroups: [""]
+    resources: ["endpoints"]
+    resourceNames: ["cluster-autoscaler"]
+    verbs: ["get", "update", "patch", "delete"]
+  # accessing & modifying cluster state (nodes & pods)
+  - apiGroups: [""]
+    resources: ["nodes"]
+    verbs: ["get", "list", "watch", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["pods"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: [""]
+    resources: ["pods/eviction"]
+    verbs: ["create"]
+  # read-only access to cluster state
+  - apiGroups: [""]
+    resources: ["services", "replicationcontrollers", "persistentvolumes", "persistentvolumeclaims"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps", "extensions"]
+    resources: ["daemonsets", "replicasets"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["apps"]
+    resources: ["statefulsets"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["policy"]
+    resources: ["poddisruptionbudgets"]
+    verbs: ["get", "list", "watch"]
+  - apiGroups: ["storage.k8s.io"]
+    resources: ["storageclasses"]
+    verbs: ["get", "list", "watch"]
+  # misc access
+  - apiGroups: [""]
+    resources: ["events"]
+    verbs: ["create", "update", "patch"]
+  - apiGroups: [""]
+    resources: ["configmaps"]
+    verbs: ["get", "list", "watch", "create", "update", "patch", "delete", "deletecollection"]
```

Comment:
cluster-wide write permission (especially deletecollection) is surprising... curious what that's for

Reply:
Cluster-Autoscaler has no API object, so it reports its status via "cluster-autoscaler-status" configmap (by default kube-system, but IIRC it's user-configurable). It deletes the configmap on shutdown. So it needs permissions to write, delete, update, etc. I don't think `deletecollection` is needed, unless I'm missing something.

Reply:
I see. yeah, remove deletecollection. can also scope the update/patch/delete permissions to a particular name "cluster-autoscaler-status" like it did for endpoints

Reply:
The reason for listing all verbs here is simply that I followed the convention from [plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go](https://github.com/kubernetes/kubernetes/blob/master/plugin/pkg/auth/authorizer/rbac/bootstrappolicy/policy.go#L31) and used ```ReadWrite``` since, as @MaciekPytel wrote, Cluster Autoscaler needs write access there.

The ```deletecollection``` is indeed superfluous but it does not actually grant anything more than the rest of verbs.

Please let me know if we have to have this removed.

Reply:
> The deletecollection is indeed superfluous but AFAIU it does not actually grant anything more than the rest of verbs.

it makes it easy to accidentally delete swaths of configmaps... let's drop it if it's unused, as we expect

Reply:
I've limited access to configmaps, as discussed above.